lib/commands/os/build-config.ts

```diff
@@ -35,10 +35,10 @@ interface ArgsDef {
 
 export default class OsBuildConfigCmd extends Command {
 	public static description = stripIndent`
-		Build an OS config and save it to a JSON file.
+		Prepare a configuration file for use by the 'os configure' command.
 
-		Interactively generate an OS config once, so that the generated config
-		file can be used in \`balena os configure\`, skipping the interactive part.
+		Interactively generate a configuration file that can then be used as
+		non-interactive input by the 'balena os configure' command.
```

Comment:
I gather that the `os build-config` command does not need to be modified because users wishing to use `os configure` non-interactively can do so with the `os configure --dev` option.

Reply:
I would say that this does deserve to have a `--developmentMode` since after all, it's something that ends up being part of the generated config.json.
In which case we probably also need to use `getOsVersionFromImage()` and error if it's s prod image.

Reply:
I agree that "development mode" may be confusing, especially given that "local mode" is really a "local development mode".

Reply:
I'm not 100% convinced yet, since `os build-config`'s job is to... build a full config that's only missing api keys etc :)
But I don't see any reason to delay the MVP for this, since if we users ask for this, then we can add it later.